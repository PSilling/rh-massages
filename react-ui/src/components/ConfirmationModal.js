// react imports
import React, { Component } from 'react';

// component imports
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom confirmation modal component.
 */
class ConfirmationModal extends Component {

  render() {
    return(
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog onClose={this.props.onClose} width="40%">
            <h3>{ _t.translate('Action confirmation') }</h3>
            <p>{this.props.message}</p>
            <ModalActions
              primaryLabel={ _t.translate('Proceed') }
              onProceed={this.props.onConfirm}
              onClose={this.props.onClose}
            />
          </ModalDialog>
      </ModalContainer>
    )
  }
}

export default ConfirmationModal
