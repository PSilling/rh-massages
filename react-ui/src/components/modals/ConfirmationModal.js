// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Col, Row, Modal, ModalBody } from "reactstrap";

// component imports
import ModalActions from "../buttons/ModalActions";

// util imports
import _t from "../../util/Translations";

/**
 * Modal with a simple action confirmation message and ModalActions.
 */
class ConfirmationModal extends Component {
  handleKeyPress = event => {
    if (event.key === "Enter" && document.activeElement.tabIndex === -1) {
      this.props.onConfirm();
    }
  };

  createInsides = () => (
    <ModalBody>
      <Row>
        <Col md="12">
          <h3>{_t.translate("Action confirmation")}</h3>
          <hr />
        </Col>
      </Row>
      <Row>
        <Col md="12">{this.props.message}</Col>
      </Row>
      <ModalActions onProceed={this.props.onConfirm} onClose={this.props.onClose}>
        {this.props.children}
      </ModalActions>
    </ModalBody>
  );

  render() {
    return this.props.withPortal ? (
      <Modal isOpen toggle={this.props.onClose} tabIndex="-1" onKeyPress={this.handleKeyPress}>
        {this.createInsides()}
      </Modal>
    ) : (
      this.createInsides()
    );
  }
}

ConfirmationModal.propTypes = {
  /** children supplied to ModalActions */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** message in the Modal */
  message: PropTypes.string.isRequired,
  /** callback function triggered on Modal close */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onConfirm: PropTypes.func.isRequired,
  /** whether ModalContainer should be used; useful for testing to avoid portals */
  withPortal: PropTypes.bool
};

ConfirmationModal.defaultProps = {
  children: null,
  withPortal: true
};

export default ConfirmationModal;
