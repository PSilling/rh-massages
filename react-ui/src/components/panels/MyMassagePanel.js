// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Row, Col, Card, Button, CardHeader, CardBody, CardText } from "reactstrap";
import moment from "moment";

// component imports
import CalendarButton from "../iconbuttons/CalendarButton";
import ConfirmationModal from "../modals/ConfirmationModal";

// util imports
import Util from "../../util/Util";
import _t from "../../util/Translations";

/**
 * Massage information panel for My Massages view.
 */
class MyMassagePanel extends Component {
  state = { active: false };

  shouldComponentUpdate(nextProps, nextState) {
    return (
      this.props.type !== nextProps.type ||
      this.props.massage.masseuse !== nextProps.massage.masseuse ||
      this.props.massage.date !== nextProps.massage.date ||
      this.props.massage.ending !== nextProps.massage.ending ||
      this.state.active !== nextState.active
    );
  }

  handleToggle = () => {
    this.setState(prevStata => ({ active: !prevStata.active }));
  };

  cancelMassage = () => {
    Util.put(
      Util.MASSAGES_URL,
      [
        {
          id: this.props.massage.id,
          date: this.props.massage.date,
          ending: this.props.massage.ending,
          masseuse: this.props.massage.masseuse,
          client: null,
          facility: this.props.massage.facility
        }
      ],
      this.props.getCallback
    );
  };

  render() {
    return (
      <div className="mb-3">
        <Row>
          <Col md="3">
            <Card color={this.props.type} outline>
              <CardHeader>
                {moment(this.props.massage.date).format("ddd L")}
                <Button
                  color="light"
                  className="close"
                  aria-label="Close"
                  onClick={this.handleToggle}
                  disabled={this.props.disabled}
                >
                  {!this.props.disabled && <span aria-hidden="true">&times;</span>}
                </Button>
              </CardHeader>
              <CardBody>
                <CardText>{`${_t.translate("Facility")}: ${this.props.massage.facility.name}`}</CardText>
                <CardText>{`${_t.translate("Masseur/Masseuse")}: ${this.props.massage.masseuse}`}</CardText>
                <CardText>
                  {`${_t.translate("Time")}: ${moment(this.props.massage.date).format("HH:mm")}â€“${moment(
                    this.props.massage.ending
                  ).format("HH:mm")}`}
                </CardText>
                <CardText>
                  {`${_t.translate("Event")}: `}
                  <CalendarButton link={Util.getEventLink(this.props.massage)} />
                </CardText>
              </CardBody>
            </Card>
          </Col>
        </Row>
        {this.state.active && (
          <ConfirmationModal
            message={_t.translate("Are you sure you want to unassign yourself from this massage?")}
            onClose={this.handleToggle}
            onConfirm={() => {
              this.handleToggle();
              this.cancelMassage();
            }}
          />
        )}
      </div>
    );
  }
}

MyMassagePanel.propTypes = {
  /** Massage to be printed inside the panel */
  massage: PropTypes.shape({
    id: PropTypes.number,
    masseuse: PropTypes.string,
    date: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
    ending: PropTypes.oneOfType([PropTypes.number, PropTypes.instanceOf(Date)]),
    client: PropTypes.shape({
      email: PropTypes.string,
      name: PropTypes.string,
      surname: PropTypes.string,
      sub: PropTypes.string,
      subscribed: PropTypes.bool
    }),
    facility: PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string
    })
  }).isRequired,
  /** type of the Bootstrap panel */
  type: PropTypes.string,
  /** whether the removal button should be hidden or not */
  disabled: PropTypes.bool,
  /** update callback function called on Massage cancellation */
  getCallback: PropTypes.func
};

MyMassagePanel.defaultProps = {
  type: "default",
  disabled: true,
  getCallback: null
};

export default MyMassagePanel;
