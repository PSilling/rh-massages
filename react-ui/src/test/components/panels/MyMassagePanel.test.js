// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import { Button, CardHeader, CardText } from "reactstrap";
import moment from "moment";
import ConfirmationModal from "../../../components/modals/ConfirmationModal";
import MyMassagePanel from "../../../components/panels/MyMassagePanel";
import _t from "../../../util/Translations";

// test mocks
jest.mock("../../../util/Util");

afterEach(() => {
  jest.resetAllMocks();
});

test("renders content with correct props", () => {
  const testFunction = jest.fn();
  const testMassage = {
    id: 1,
    date: new Date(0),
    ending: new Date(1000),
    client: null,
    facility: { id: 1, name: "test" }
  };
  const wrapper = shallow(<MyMassagePanel massage={testMassage} getCallback={testFunction} />);
  const texts = wrapper.find(CardText);
  const header = wrapper.find(CardHeader);
  const button = wrapper.find(Button);

  expect(header.props().children[0]).toEqual(moment(testMassage.date).format("ddd L"));
  expect(texts.get(0).props.children).toEqual(`${_t.translate("Facility")}: ${testMassage.facility.name}`);
  expect(texts.get(1).props.children).toEqual(`${_t.translate("Masseur/Masseuse")}: ${testMassage.masseuse}`);
  expect(texts.get(2).props.children).toEqual(
    `${_t.translate("Time")}: ${moment(testMassage.date).format("HH:mm")}â€“${moment(testMassage.ending).format("HH:mm")}`
  );
  expect(testFunction).not.toHaveBeenCalled();

  button.props().onClick();

  const modal = wrapper.find(ConfirmationModal);

  modal.props().onConfirm();

  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(wrapper).toMatchSnapshot();
});
