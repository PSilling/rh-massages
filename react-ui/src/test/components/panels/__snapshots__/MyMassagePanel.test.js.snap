// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyMassagePanel
    disabled={true}
    getCallback={[Function]}
    massage={
        Object {
            "client": null,
            "date": 1970-01-01T00:00:00.000Z,
            "ending": 1970-01-01T00:00:01.000Z,
            "facility": Object {
              "id": 1,
              "name": "test",
            },
            "id": 1,
          }
    }
    type="default"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Row
          tag="div"
>
          <Col
                    md="3"
                    tag="div"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    <Card
                              color="default"
                              outline={true}
                              tag="div"
                    >
                              <CardHeader
                                        tag="div"
                              >
                                        Thu 01/01/1970
                                        <Button
                                                  aria-label="Close"
                                                  className="close"
                                                  color="light"
                                                  disabled={true}
                                                  onClick={[Function]}
                                                  tag="button"
                                        />
                              </CardHeader>
                              <CardBody
                                        tag="div"
                              >
                                        <CardText
                                                  tag="p"
                                        >
                                                  Budova: test
                                        </CardText>
                                        <CardText
                                                  tag="p"
                                        >
                                                  Masér/Masérka: undefined
                                        </CardText>
                                        <CardText
                                                  tag="p"
                                        >
                                                  Čas: 01:00–01:00
                                        </CardText>
                                        <CardText
                                                  tag="p"
                                        >
                                                  Událost: 
                                                  <CalendarButton
                                                            link=""
                                                  />
                                        </CardText>
                              </CardBody>
                    </Card>
          </Col>
</Row>,
        false,
      ],
      "className": "mb-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            md="3"
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <Card
                        color="default"
                        outline={true}
                        tag="div"
            >
                        <CardHeader
                                    tag="div"
                        >
                                    Thu 01/01/1970
                                    <Button
                                                aria-label="Close"
                                                className="close"
                                                color="light"
                                                disabled={true}
                                                onClick={[Function]}
                                                tag="button"
                                    />
                        </CardHeader>
                        <CardBody
                                    tag="div"
                        >
                                    <CardText
                                                tag="p"
                                    >
                                                Budova: test
                                    </CardText>
                                    <CardText
                                                tag="p"
                                    >
                                                Masér/Masérka: undefined
                                    </CardText>
                                    <CardText
                                                tag="p"
                                    >
                                                Čas: 01:00–01:00
                                    </CardText>
                                    <CardText
                                                tag="p"
                                    >
                                                Událost: 
                                                <CalendarButton
                                                            link=""
                                                />
                                    </CardText>
                        </CardBody>
            </Card>
</Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Card
              color="default"
              outline={true}
              tag="div"
>
              <CardHeader
                            tag="div"
              >
                            Thu 01/01/1970
                            <Button
                                          aria-label="Close"
                                          className="close"
                                          color="light"
                                          disabled={true}
                                          onClick={[Function]}
                                          tag="button"
                            />
              </CardHeader>
              <CardBody
                            tag="div"
              >
                            <CardText
                                          tag="p"
                            >
                                          Budova: test
                            </CardText>
                            <CardText
                                          tag="p"
                            >
                                          Masér/Masérka: undefined
                            </CardText>
                            <CardText
                                          tag="p"
                            >
                                          Čas: 01:00–01:00
                            </CardText>
                            <CardText
                                          tag="p"
                            >
                                          Událost: 
                                          <CalendarButton
                                                        link=""
                                          />
                            </CardText>
              </CardBody>
</Card>,
            "md": "3",
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <CardHeader
                  tag="div"
>
                  Thu 01/01/1970
                  <Button
                                    aria-label="Close"
                                    className="close"
                                    color="light"
                                    disabled={true}
                                    onClick={[Function]}
                                    tag="button"
                  />
</CardHeader>,
                <CardBody
                  tag="div"
>
                  <CardText
                                    tag="p"
                  >
                                    Budova: test
                  </CardText>
                  <CardText
                                    tag="p"
                  >
                                    Masér/Masérka: undefined
                  </CardText>
                  <CardText
                                    tag="p"
                  >
                                    Čas: 01:00–01:00
                  </CardText>
                  <CardText
                                    tag="p"
                  >
                                    Událost: 
                                    <CalendarButton
                                                      link=""
                                    />
                  </CardText>
</CardBody>,
              ],
              "color": "default",
              "outline": true,
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Thu 01/01/1970",
                    <Button
                      aria-label="Close"
                      className="close"
                      color="light"
                      disabled={true}
                      onClick={[Function]}
                      tag="button"
/>,
                  ],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  "Thu 01/01/1970",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "aria-label": "Close",
                      "children": false,
                      "className": "close",
                      "color": "light",
                      "disabled": true,
                      "onClick": [Function],
                      "tag": "button",
                    },
                    "ref": null,
                    "rendered": false,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <CardText
                      tag="p"
>
                      Budova: test
</CardText>,
                    <CardText
                      tag="p"
>
                      Masér/Masérka: undefined
</CardText>,
                    <CardText
                      tag="p"
>
                      Čas: 01:00–01:00
</CardText>,
                    <CardText
                      tag="p"
>
                      Událost: 
                      <CalendarButton
                                            link=""
                      />
</CardText>,
                  ],
                  "tag": "div",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Budova: test",
                      "tag": "p",
                    },
                    "ref": null,
                    "rendered": "Budova: test",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Masér/Masérka: undefined",
                      "tag": "p",
                    },
                    "ref": null,
                    "rendered": "Masér/Masérka: undefined",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "Čas: 01:00–01:00",
                      "tag": "p",
                    },
                    "ref": null,
                    "rendered": "Čas: 01:00–01:00",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "Událost: ",
                        <CalendarButton
                          link=""
/>,
                      ],
                      "tag": "p",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Událost: ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "link": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Row
            tag="div"
>
            <Col
                        md="3"
                        tag="div"
                        widths={
                                    Array [
                                                "xs",
                                                "sm",
                                                "md",
                                                "lg",
                                                "xl",
                                              ]
                        }
            >
                        <Card
                                    color="default"
                                    outline={true}
                                    tag="div"
                        >
                                    <CardHeader
                                                tag="div"
                                    >
                                                Thu 01/01/1970
                                                <Button
                                                            aria-label="Close"
                                                            className="close"
                                                            color="light"
                                                            disabled={true}
                                                            onClick={[Function]}
                                                            tag="button"
                                                />
                                    </CardHeader>
                                    <CardBody
                                                tag="div"
                                    >
                                                <CardText
                                                            tag="p"
                                                >
                                                            Budova: test
                                                </CardText>
                                                <CardText
                                                            tag="p"
                                                >
                                                            Masér/Masérka: undefined
                                                </CardText>
                                                <CardText
                                                            tag="p"
                                                >
                                                            Čas: 01:00–01:00
                                                </CardText>
                                                <CardText
                                                            tag="p"
                                                >
                                                            Událost: 
                                                            <CalendarButton
                                                                        link=""
                                                            />
                                                </CardText>
                                    </CardBody>
                        </Card>
            </Col>
</Row>,
          false,
        ],
        "className": "mb-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              md="3"
              tag="div"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              <Card
                            color="default"
                            outline={true}
                            tag="div"
              >
                            <CardHeader
                                          tag="div"
                            >
                                          Thu 01/01/1970
                                          <Button
                                                        aria-label="Close"
                                                        className="close"
                                                        color="light"
                                                        disabled={true}
                                                        onClick={[Function]}
                                                        tag="button"
                                          />
                            </CardHeader>
                            <CardBody
                                          tag="div"
                            >
                                          <CardText
                                                        tag="p"
                                          >
                                                        Budova: test
                                          </CardText>
                                          <CardText
                                                        tag="p"
                                          >
                                                        Masér/Masérka: undefined
                                          </CardText>
                                          <CardText
                                                        tag="p"
                                          >
                                                        Čas: 01:00–01:00
                                          </CardText>
                                          <CardText
                                                        tag="p"
                                          >
                                                        Událost: 
                                                        <CalendarButton
                                                                      link=""
                                                        />
                                          </CardText>
                            </CardBody>
              </Card>
</Col>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Card
                color="default"
                outline={true}
                tag="div"
>
                <CardHeader
                                tag="div"
                >
                                Thu 01/01/1970
                                <Button
                                                aria-label="Close"
                                                className="close"
                                                color="light"
                                                disabled={true}
                                                onClick={[Function]}
                                                tag="button"
                                />
                </CardHeader>
                <CardBody
                                tag="div"
                >
                                <CardText
                                                tag="p"
                                >
                                                Budova: test
                                </CardText>
                                <CardText
                                                tag="p"
                                >
                                                Masér/Masérka: undefined
                                </CardText>
                                <CardText
                                                tag="p"
                                >
                                                Čas: 01:00–01:00
                                </CardText>
                                <CardText
                                                tag="p"
                                >
                                                Událost: 
                                                <CalendarButton
                                                                link=""
                                                />
                                </CardText>
                </CardBody>
</Card>,
              "md": "3",
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <CardHeader
                    tag="div"
>
                    Thu 01/01/1970
                    <Button
                                        aria-label="Close"
                                        className="close"
                                        color="light"
                                        disabled={true}
                                        onClick={[Function]}
                                        tag="button"
                    />
</CardHeader>,
                  <CardBody
                    tag="div"
>
                    <CardText
                                        tag="p"
                    >
                                        Budova: test
                    </CardText>
                    <CardText
                                        tag="p"
                    >
                                        Masér/Masérka: undefined
                    </CardText>
                    <CardText
                                        tag="p"
                    >
                                        Čas: 01:00–01:00
                    </CardText>
                    <CardText
                                        tag="p"
                    >
                                        Událost: 
                                        <CalendarButton
                                                            link=""
                                        />
                    </CardText>
</CardBody>,
                ],
                "color": "default",
                "outline": true,
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Thu 01/01/1970",
                      <Button
                        aria-label="Close"
                        className="close"
                        color="light"
                        disabled={true}
                        onClick={[Function]}
                        tag="button"
/>,
                    ],
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Thu 01/01/1970",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "aria-label": "Close",
                        "children": false,
                        "className": "close",
                        "color": "light",
                        "disabled": true,
                        "onClick": [Function],
                        "tag": "button",
                      },
                      "ref": null,
                      "rendered": false,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <CardText
                        tag="p"
>
                        Budova: test
</CardText>,
                      <CardText
                        tag="p"
>
                        Masér/Masérka: undefined
</CardText>,
                      <CardText
                        tag="p"
>
                        Čas: 01:00–01:00
</CardText>,
                      <CardText
                        tag="p"
>
                        Událost: 
                        <CalendarButton
                                                link=""
                        />
</CardText>,
                    ],
                    "tag": "div",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Budova: test",
                        "tag": "p",
                      },
                      "ref": null,
                      "rendered": "Budova: test",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Masér/Masérka: undefined",
                        "tag": "p",
                      },
                      "ref": null,
                      "rendered": "Masér/Masérka: undefined",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "Čas: 01:00–01:00",
                        "tag": "p",
                      },
                      "ref": null,
                      "rendered": "Čas: 01:00–01:00",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": Array [
                          "Událost: ",
                          <CalendarButton
                            link=""
/>,
                        ],
                        "tag": "p",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Událost: ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "link": "",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                      ],
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
